### PR TITLE
Complete symbol loader on import of toplevel symbol

### LINKED ISSUE #12736: Non-visible symbol in wildcard import taking precedence over visible symbol in current package
## Reproduction steps

Scala version: 2.13.10

https://github.com/armanbilge/sandbox/commit/0c6533cff815c823c13dd7397289ebc49edf0526

## Problem

```
sbt:sandbox> bar/compile
[info] compiling 1 Scala source to /workspace/sandbox/foo/target/scala-2.13/classes ...
[info] compiling 2 Scala sources to /workspace/sandbox/bar/target/scala-2.13/classes ...
[error] /workspace/sandbox/bar/src/main/scala/bar.scala:7:3: object Foo in package foo cannot be accessed as a member of package foo from object Bar in package baz
[error]   Foo
[error]   ^
[error] one error found
[error] (bar / Compile / compileIncremental) Compilation failed
[error] Total time: 0 s, completed Feb 22, 2023, 7:15:12 AM
sbt:sandbox> 
```

Meanwhile, there is a visible `Foo` in the current package, which is the one we want.

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#12736

### CODE REVIEW

**Comment by SethTisue** [comment]
let's see what the community build thinks of 2.13.11-bin-e744d0b-SNAPSHOT: https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/4322/ queued

File: src/compiler/scala/tools/nsc/typechecker/Contexts.scala [hunk 1]
```diff
@@ -1248,7 +1248,10 @@ trait Contexts { self: Analyzer =>
      */
     private[Contexts] def importedAccessibleSymbol(imp: ImportInfo, sym: => Symbol): Symbol =
       if (isExcludedRootImport(imp)) NoSymbol
-      else sym.filter(isAccessible(_, imp.qual.tpe, superAccess = false))
+      else sym.filter { s =>
+        if (s.isTopLevel) s.initialize
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/typechecker/Contexts.scala]: Forcing makes me a bit nervous..

We could call `sym.exists` instead, which only completes symbol loaders, not lazy types. Though it also completes `UnPickler$Scan$LazyTypeRef`, not sure why that is. IIUC, we only would need to complete `SymbolLoader`s.

In the test case we get `SymbolLoaders$ClassfileLoader` --(.exists)--> `UnPickler$Scan$LazyTypeRef` --(.exists)--> (`Types$ModuleTypeRef`).

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/typechecker/Contexts.scala]: This was inspired by your `exists` trick, I understood the completer, but thought `initialize` was equivalent. I can try detuning to `exists`. And check whether it matters what is imported.

**Comment by SethTisue** [comment]
2.13.11-bin-e744d0b-SNAPSHOT / https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/4330/ 

* metals failure is expected (breaks on any PR validation snapshot)
* scala-java8-compat failure is shared with #10333 so before worrying about it let’s see if it also shows up at https://github.com/scala/community-build/pull/1646 , maybe some recently merged PR is the culprit
* squants is failing with "illegal cyclic reference involving class Dimensionless" — again, not sure if this PR is the actual culprit

**Comment by som-snytt** [comment]
Sorry, just realized I've been wearing my "May the Force Be With You. Always." t-shirt under my sweatshirt this whole time.

I'll try squants locally and with the lrytz `exists` recommendation.

**Comment by som-snytt** [comment]
Confirmed that squants is happy with the amended fix, which I will push now. Thanks @lrytz and @SethTisue .

File: src/compiler/scala/tools/nsc/typechecker/Contexts.scala (hunk 1 above)
Comment (lrytz) [reply src/compiler/scala/tools/nsc/typechecker/Contexts.scala]: Here's a spot in the compiler where we call `load` twice: https://github.com/scala/scala/blob/0fd61f7eae54ae79b0713a40d8f2ef9e0962064f/src/reflect/scala/reflect/internal/Symbols.scala#L1799-L1805

**Comment by SethTisue** [comment]
> scala-java8-compat failure is shared with https://github.com/scala/scala/pull/10333 so before worrying about it let’s see if it also shows up at https://github.com/scala/community-build/pull/1646 , maybe some recently merged PR is the culprit

scala-java8-compat did not fail on my PR

as I also suggested over on your other PR, I guess try rebasing onto 0fd61f7 which is the new community build baseline

**Comment by SethTisue** [comment]
Oh, the scala-java8-compat test in question (`Issue247Test`) is an extremely weird test — it uses coursier to retrieve artifacts at runtime. I'll just disable it in the community build.

**Comment by som-snytt** [comment]
Rebased, and `isTopLevel` check is redundant before `exists`.

Restored to 2.13.11 because the status quo is poor. In particular, people are somewhat confused by top-level `private object` or `private[p] object` because sometimes it works and sometimes it doesn't. This bug does not belong in the club of esoteric "can we reach a decision on access" tickets.

**Review by lrytz — APPROVED** [review]
👍  thank you!

**Comment by som-snytt** [comment]
Thanks for the boosted comment. I'd note that braces are not required to group a line comment.

**Pull request merged**